src/app/js/formats/istexSummary/getIstexData.js

```diff
@@ -103,100 +148,126 @@ export const parseOtherData = ({ response, error }) => {
     };
 };
 
-export const getOtherVolumeData = ({ issn, year, searchedField }) =>
+export const getOtherVolumeData = ({ value, year, searchedField }) =>
     composeAsync(
-        getOtherVolumeUrl({ issn, year, searchedField }),
+        getOtherVolumeUrl({ value, year, searchedField }),
         fetch,
         parseOtherData,
     );
 
-export const addOtherVolumeData = ({ issn, year, searchedField }) => async ({
+export const addOtherVolumeData = ({ value, year, searchedField }) => async ({
     hits,
 }) => ({
-    hits: [...hits, await getOtherVolumeData({ issn, year, searchedField })()],
+    hits: [...hits, await getOtherVolumeData({ value, year, searchedField })()],
 });
 
-export const getVolumeData = ({ issn, year, searchedField }) =>
+export const getVolumeData = ({ value, year, searchedField }) =>
     composeAsync(
-        getVolumeUrl({ issn, year, searchedField }),
+        getVolumeUrl({ value, year, searchedField }),
         fetch,
         parseVolumeData,
-        addOtherVolumeData({ issn, year, searchedField }),
+        addOtherVolumeData({ value, year, searchedField }),
     );
 
 const getVolumeQuery = volume =>
     volume === 'other' ? '-host.volume:[0 TO *]' : `host.volume:"${volume}"`;
 
-export const getIssueUrl = ({ issn, year, volume, searchedField }) => () => ({
-    url: `${ISTEX_API_URL}/document/?q=(${encodeURIComponent(
-        `${searchedField}:"${issn}" AND publicationDate:"${year}" AND ${getVolumeQuery(
-            volume,
-        )}`,
-    )})&facet=host.issue[*-*:1]&size=0&output=*`,
+export const getIssueUrl = ({ value, year, volume, searchedField }) => () => ({
+    url: buildIstexQuery({
+        query: `${getFilterQuery(
```

Comment:
How about having getFilterQuery construct the whole filter query. With additional publicationDate, volume and issue arg?

Reply:
Would be a nice addition if we change the code later.